starknet_py/net/models/typed_data.py

```diff
@@ -14,14 +15,24 @@ class Parameter(TypedDict):
     type: str
 
 
-class StarkNetDomain(TypedDict):
+class Revision(Enum):
     """
-    TypedDict representing a StarkNetDomain object
+    Enum representing the revision of the specification to be used.
+    """
+
+    V0 = 0
+    V1 = 1
+
+
+class Domain(TypedDict):
+    """
+    TypedDict representing a domain object (both StarkNetDomain, StarknetDomain)
     """
 
     name: str
     version: str
     chainId: Union[str, int]
+    revision: Optional[Union[str, int]]
```

Comment:
We could also use `Optional[Revision]`, don't have a strong opinion here, wdyt @ddoktorski ?